redash/query_runner/impala_ds.py

```diff
@@ -95,10 +96,12 @@ def run_query(self, query, user):
 
         connection = None
         try:
-            connection = connect(**self.configuration.to_dict())
-
-            cursor = connection.cursor()
-
+            config = self.configuration.to_dict()
+            connection = connect(**config)
+            if 'user' in config:
+                cursor = connection.cursor(user=config['user'])
+            else:
+                cursor = connection.cursor()
```

Comment:
What happens if there are both `ldap_user` and `user` fields? Does this fail gracefully?

Reply:
@susodapop  hi there, I update the code~